### PR TITLE
Feature/LP169 community validations

### PR DESCRIPTION
**Pull request by Lupillo17**
Added validations to the viewmodel when retrieving data from repository. 
Actually exists only 3 states, succes, loading and error.

State succes:
image

State Loading
image

State Error
image

### CODE REVIEW

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityViewModel.kt [hunk 1]
```diff
@@ -44,9 +44,11 @@ class CommunityViewModel : ViewModel() {
                 _communitySections.clear()
                 _communitySections.addAll(response.data)
             }
+
```
Comment (carlosph816) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityViewModel.kt:49]: Let's delete this space please

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityViewModel.kt [hunk 2]
```diff
@@ -44,9 +44,11 @@ class CommunityViewModel : ViewModel() {
...
             is DataState.Error -> {
                 _state.emit(State.Error)
             }
+
```
Comment (carlosph816) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityViewModel.kt:53]: Let's delete this space please

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt [hunk 3]
```diff
@@ -97,19 +112,24 @@ fun CommunityView(
                             CommunityCardType.Amenities -> {
                                 navigationToAmenities()
                             }
+
```
Comment (carlosph816) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: Let's delete this space please

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt [hunk 4]
```diff
@@ -97,19 +112,24 @@ fun CommunityView(
...
                             CommunityCardType.Events -> {
                                 /* TODO() This view does not exist, as soon as
                                     this view is defined the navigation of this section is added */
                             }
+
```
Comment (carlosph816) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: Let's delete this space please

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt [hunk 5]
```diff
@@ -97,19 +112,24 @@ fun CommunityView(
...
                             CommunityCardType.CommunityWall -> {
                                 navigationToCommunityWall()
                             }
+
```
Comment (carlosph816) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: Let's delete this space please

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt [hunk 6]
```diff
@@ -97,19 +112,24 @@ fun CommunityView(
...
                             CommunityCardType.DoItYourself -> {
                                 /* TODO() This view does not exist, as soon as
                                     this view is defined the navigation of this section is added */
                             }
+
```
Comment (carlosph816) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: Let's delete this space please

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt [hunk 7]
```diff
@@ -73,12 +75,25 @@ fun CommunityView(
                 end = sizes.midSmallSize
             )
     ) {
+        if (state.value != State.Success) {
```
Comment (irv205) [inline feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: Change if to when if you want to manage more states

Comment (Lupillo17) [reply feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: solving

File: feature/community/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -7,4 +7,6 @@
     <string name="cardButton_services">Services</string>
     <string name="title_community">Community</string>
     <string name="default_card_description"> </string>
+    <string name="error_message_retrieving_repository_data">Error</string>
```
Comment (carlosph816) [inline feature/community/src/main/res/values/strings.xml]: I think just with the name error_message is okay

File: feature/community/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -7,4 +7,6 @@
...
+    <string name="loading_message_retrieving_repository_data">Loading</string>
```
Comment (carlosph816) [inline feature/community/src/main/res/values/strings.xml]: I think just with the name loading_message is okay

File: feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt (hunk 7 above)
Comment (Lupillo17) [reply feature/community/src/main/java/com/iteneum/community/presentation/CommunityView.kt]: solved

File: feature/community/src/main/res/values/strings.xml (hunk 8 above)
Comment (Lupillo17) [reply feature/community/src/main/res/values/strings.xml]: solved

File: feature/community/src/main/res/values/strings.xml (hunk 9 above)
Comment (Lupillo17) [reply feature/community/src/main/res/values/strings.xml]: solved

**Review by carlosph816 — APPROVED** [review]

**Review by irv205 — APPROVED** [review]

**Pull request merged**